src/search-headless.ts

```diff
@@ -0,0 +1,549 @@
+   */
+  async executeUniversalQuery(): Promise<UniversalSearchResponse | undefined> {
+    if (this.state.meta.searchType !== SearchTypeEnum.Universal) {
+      console.error('The meta.searchType must be set to \'universal\' for universal search. '
+        + 'Set the searchType to universal by calling `setUniversal()`');
+      return;
+    }
+    const thisRequestId = this.httpManager.updateRequestId('universalQuery');
+    this.stateManager.dispatchEvent('searchStatus/setIsLoading', true);
+    const { input, querySource, queryTrigger } = this.state.query;
+    const skipSpellCheck = !this.state.spellCheck.enabled;
+    const sessionTrackingEnabled = this.state.sessionTracking.enabled;
+    const { limit, restrictVerticals } = this.state.universal;
+    const sessionId = this.state.sessionTracking.sessionId;
+    const { referrerPageUrl, context } = this.state.meta;
+    const { userLocation } = this.state.location;
+
+    const request = {
+      query: input || '',
+      querySource,
+      queryTrigger,
+      skipSpellCheck,
+      ...(sessionTrackingEnabled && { sessionId }),
+      sessionTrackingEnabled,
+      limit,
+      location: userLocation,
+      context,
+      referrerPageUrl,
+      restrictVerticals,
+      additionalHttpHeaders: this.additionalHttpHeaders
+    };
+
+    let response: UniversalSearchResponse;
+    try {
+      response = await this.core.universalSearch(request);
+    } catch (e) {
+      const isLatestResponse = this.httpManager.processRequestId('universalQuery', thisRequestId);
+      if (isLatestResponse) {
+        this.stateManager.dispatchEvent('searchStatus/setIsLoading', false);
+      }
+      return Promise.reject(e);
+    }
+    const isLatestResponse = this.httpManager.processRequestId('universalQuery', thisRequestId);
+    if (!isLatestResponse) {
+      return response;
+    }
+    this.stateManager.dispatchEvent('universal/setVerticals', response.verticalResults);
+    this.stateManager.dispatchEvent('query/setQueryId', response.queryId);
+    this.stateManager.dispatchEvent('query/setMostRecentSearch', input);
+    this.stateManager.dispatchEvent('spellCheck/setResult', response.spellCheck);
+    this.stateManager.dispatchEvent('query/setSearchIntents', response.searchIntents || []);
+    this.stateManager.dispatchEvent('location/setLocationBias', response.locationBias);
+    this.stateManager.dispatchEvent('searchStatus/setIsLoading', false);
+    this.stateManager.dispatchEvent('meta/setUUID', response.uuid);
+    this.stateManager.dispatchEvent('directAnswer/setResult', response.directAnswer);
+    this.stateManager.dispatchEvent('queryRules/setActions', response.queryRulesActionsData || []);
+    return response;
+  }
+
+  /**
+   * Performs an autocomplete request across all verticals using the query input
+   * stored in state.
+   *
+   * @returns A Promise of an {@link AutocompleteResponse} from the Answers API
```

Comment:
can we update the docs for all the functions in this exported class?

Reply:
I just need to `run build` again to do that right?